### DISCUSSION TITLE
assigning default variable values

### DISCUSSION THREAD

**Discussion by mundorfd** [ttscoff/bunch-docs#344] [Q&A]
I am having trouble assigning and using variables.

here is my test script:

```
---
title: test
---

${myname:John}

(log "myname is set to ${myname}")

myname = "Deb"

(log "myname is set to ${myname}")
```
and here is the log output

```
2024-12-10 09:04:50: ⚪[test] Open >>>>>>>>>>>>>>>>>>>>>
2024-12-10 09:04:50: ⚪[test] Launching app (Deb)
2024-12-10 09:04:50: ⚪[test] Processing command "log "myname is set to Deb""
2024-12-10 09:04:50: 🟡[test] Logger: "myname is set to Deb"
2024-12-10 09:04:50: ⚪[test] Processing command "log "myname is set to Deb""
2024-12-10 09:04:50: 🟡[test] Logger: "myname is set to Deb"
```

Not sure why the variable isn't set to "John" in the first logging, and why is it trying to launch an app?

**Comment by mundorfd** [comment]
Is this forum still active?

**Comment by ttscoff** [comment]
It is, and I have no idea why my response never showed up here. Here's what I wrote back on Dec 10:

Variable assignments are not sequential. It's a shortcoming of the syntax I created. The entire file is processed before acting, so setting a variable at the end that you call at the beginning will output the value from the end. John isn't showing up because there's a value for $myname. I really should try to fix this, but it's a deep issue that would require some major refactoring (like an entire overhaul). When I originally implemented variables it was assumed that you would want to set them once via a script or frontmatter, and that they should be available to the entire document.
The reason it's trying to launch an app is because you have a variable on a line with nothing else, which Bunch reads as an app command. In this case you just have "Deb" on a line, so it assumes that's an app name.
